components/script/script_thread.rs

```diff
@@ -1606,31 +1532,31 @@ impl ScriptThread {
         // TODO: The specification says to filter out non-renderable documents,
         // as well as those for which a rendering update would be unnecessary,
         // but this isn't happening here.
-        let pipelines_to_update: Vec<PipelineId> = self
-            .documents
-            .borrow()
-            .iter()
-            .filter(|(_, document)| document.window().is_top_level())
-            .flat_map(|(id, document)| {
-                std::iter::once(id).chain(
-                    document
-                        .iter_iframes()
-                        .filter_map(|iframe| iframe.pipeline_id()),
-                )
-            })
-            .collect();
+
+        // > 2. Let docs be all fully active Document objects whose relevant agent's event loop
+        // > is eventLoop, sorted arbitrarily except that the following conditions must be
+        // > met:
+        //
+        // > Any Document B whose container document is A must be listed after A in the
+        // > list.
+        //
+        // > If there are two documents A and B that both have the same non-null container
+        // > document C, then the order of A and B in the list must match the
+        // > shadow-including tree order of their respective navigable containers in C's
+        // > node tree.
+        //
+        // > In the steps below that iterate over docs, each Document must be processed in
+        // > the order it is found in the list.
+        let documents_in_order = self.documents.borrow().documents_in_order();
```

Comment:
I don't understand which documents we would miss before. Are we talking about cross origin iframes? Should there rendering not be updated in the script-thread managing them?

Reply:
Here's an example of a document we should miss:

```
Top Level Frame (Script Thread 1)
  - IFrame1 (Script Thread 2)
  - IFrame2 (Script Thread 2)
```

We would miss `Iframe1` and `Iframe2` in this case because they aren't top-level frames and they also aren't children of any Documents managed by `Script Thread 2`.

Reply:
Ok, thanks for the explanation.